rustler_mix/lib/rustler/compiler/config.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Rustler.Compiler.Config do
+  @moduledoc false
+
+  @type rust_version :: :stable | :beta | :nightly | binary()
+  @type cargo :: :system | {:rustup, rust_version()} | {:bin, Path.t()}
+  @type crate :: atom()
+  @type default_features :: boolean()
+  @type env :: [{binary(), binary()}]
+  @type features :: [binary()]
+  @type mode :: :debug | :release
+  @type load_data :: term()
+  @type path :: Path.t()
+
+  defstruct cargo: :system,
+            crate: nil,
+            default_features: true,
+            env: [],
+            external_resources: [],
+            features: [],
+            load_data: 0,
+            load_from: nil,
+            mode: :release,
+            otp_app: nil,
+            path: "",
+            priv_dir: "",
+            skip_compilation?: false,
+            target: nil,
+            target_dir: ""
+
+  alias Rustler.Compiler.Config
+
+  def from(otp_app, module, opts) do
+    config = Application.get_env(otp_app, module, [])
+
+    cargo = Keyword.get(config, :cargo, :system)
+    crate = config[:crate] || opts[:crate] || otp_app
+    default_features = Keyword.get(config, :default_features, true)
+    env = Keyword.get(config, :env, [])
+    features = Keyword.get(config, :features, [])
+    mode = config[:mode] || opts[:mode] || build_mode(Mix.env())
+    load_data = config[:load_data] || opts[:load_data] || 0
+
+    load_from =
+      case config[:load_from] do
+        {_, _} -> config[:load_from]
+        _ -> {otp_app, "priv/native/lib#{crate}"}
+      end
+
+    path = config[:path] || "native/#{crate}"
```

Comment:
This is missing `|| opts[:path]`

Reply:
Maybe we could simplify this function to avoid such errors:

```elixir
config = Keyword.merge(config, opts)
defaults = [default_features: true, env: [], features: [], mode: build_mode(Mix.env()), load_data: 0, ...]
merged = Keyword.merge(defaults, config)
struct!(Config, merged)
```

Reply:
We could even get the defaults from the config definition itself:

```elixir
defaults = Map.from_struct(%Config{})
```

Reply:
the `opts` is for users of your NIF lib to be able to configure various compile-time aspects of things...I can't think of a good reason why anyone would need to point the path to somewhere else, do you?